V3-Upgrade-Guide.md

```diff
@@ -97,80 +75,68 @@ Old example, not working anymore:
 ```
 
 New example:
+
 ```json
 {
   "myCustomField": "myCustomValue"
 }
 ```
 
-##### `ODataRequestConfig` class
-
-The constructor of `ODataRequestConfig` was changed so that the third parameter cannot be a `string` anymore.
-Passing in a string which was then interpreted as the value for the `Content-Type` HTTP header was deprecated.
-The type of the parameter is now `Record<string, any>`, and if only want to set the `Content-Type` HTTP header you can do so by passing `{'content-type': 'some-value'}` to the constructor.
-
-<!-- TODO: This is only meant as an example for sections in the upgrade guide. Improve this section and add new sections as you see fit.
-
-### Generator CLI
+- "Content-type" HTTP headers cannot be passed as a string in the constructor of `ODataRequestConfig` anymore. Instead pass an object with "content-type" as a key and the header value as a value, e.g. `{ 'content-type': 'some-value' }` to the constructor.
 
-The SAP Cloud SDK includes two "generator" cli applications for OData and for OpenAPI clients.
-For historic reasons the command-line arguments of both applications were different in cases where this does not make sense.
-In version 3, the arguments are aligned and deprecated arguments have been removed.
-Please see (insert link here) for the current documentation on the cli arguments.
--->
+## Switch to Middlewares for Timeouts
 
-### Timeout 
-
-The `timeout()` method was removed from the request builder and the `timeout` option was removed from the `executeHttpRequest()` function. 
+The `timeout()` method was removed from the request builder and the `timeout` option was removed from the `executeHttpRequest()` function.
 If you want to set a timeout for a request use the new timeout middleware:
 
 ```ts
-import { timeout} from '@sap-cloud-sdk/resilience'
+import { timeout } from '@sap-cloud-sdk/resilience';
 
- executeHttpRequest(myDestination, {
-          method: 'get',          
-          middleware: [timeout()]
-        })
+executeHttpRequest(myDestination, {
+  method: 'get',
+  middleware: [timeout()]
+});
 
-myRequestBuilder
-    .getAll()
-    .middleware([timeout()])
-    .execute(myDestination);
+myRequestBuilder.getAll().middleware([timeout()]).execute(myDestination);
 ```
 
-You find a detailed guide on the general [middleware concept](https://sap.github.io/cloud-sdk/docs/js/v3/features/middleware)  and the [resilience middlewares](https://sap.github.io/cloud-sdk/docs/js/v3/guides/resilience) in particular on the documentation portal.
+You find a detailed guide on the general [middleware concept](https://sap.github.io/cloud-sdk/docs/js/v3/features/middleware) and the [resilience middlewares](https://sap.github.io/cloud-sdk/docs/js/v3/guides/resilience) in particular on the documentation portal.
```

Comment:
take care about those `v3` links, if I remember correctly we'll remove that path component once the v3 is released, right?

Reply:
Yes, we should

Reply:
hm, @deekshas8 also commented on this - I must have forgotten to change it at all places.

Reply:
> version 3 beeing selected and not the latest version.

yes my understanding is that 'latest' will change once we do the release which will then also change the urls.. but I think we can also keep the v3 url paths.